src/Nat.mo

```diff
@@ -1,72 +1,299 @@
-/// Natural numbers
+/// Utility functions for working with natural numbers.
```

Comment:
Maybe it is useful to mention that the (unsigned) numbers have infinite precision, similar to `Int.mo`.